### PR TITLE
Secure information stored in SharedPreferences

### PR DESCRIPTION
**Pull request by kvasukib**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-core/src/main/java/com/amazonaws/internal/keyvaluestore/AWSKeyValueStore.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                final String actualKey = key + SHARED_PREFERENCES_DATA_IDENTIFIER_SUFFIX;
+
+                // If data not found for the key, return null.
+                if (!sharedPreferences.contains(actualKey)) {
+                    return null;
+                }
+
+                // If the version of data stored mismatches with the version of the store,
+                // return null.
+                final int keyValueStoreVersion = Integer.parseInt(
```
Comment (minbi) [inline aws-android-sdk-core/src/main/java/com/amazonaws/internal/keyvaluestore/AWSKeyValueStore.java]: try catch?

File: aws-android-sdk-core/src/main/java/com/amazonaws/util/TimingInfoUnmodifiable.java [hunk 2]
```diff
@@ -36,15 +36,7 @@
      * Always throws {@link UnsupportedOperationException}.
      */
     @Override
-    public void setEndTime(long _) {
```
Comment (minbi) [inline aws-android-sdk-core/src/main/java/com/amazonaws/util/TimingInfoUnmodifiable.java]: API removed?

File: aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java [hunk 3]
```diff
@@ -353,13 +363,30 @@ public void run() {
                             callback.onResult(getUserStateDetails(true));
                             return;
                         }
+
+                        // Read persistence key from the awsconfiguration.json and set the flag
+                        // appropriately.
+                        try {
+                            if (awsConfiguration.optJsonObject("Auth") != null &&
+                                    awsConfiguration.optJsonObject("Auth").has("Persistence")) {
+                                mIsPersistenceEnabled = awsConfig
```
Comment (minbi) [inline aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java]: awsConfiguration

Comment (kvasukib) [reply aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java]: Fixed.

**Pull request merged**